plugins/inputs/vsphere/client.go

```diff
@@ -191,3 +211,43 @@ func (c *Client) GetServerTime(ctx context.Context) (time.Time, error) {
 	}
 	return *t, nil
 }
+
+// GetMaxQueryMetrics returns the max_query_metrics setting as configured in vCenter
+func (c *Client) GetMaxQueryMetrics(ctx context.Context) (int, error) {
+	ctx, cancel := context.WithTimeout(ctx, c.Timeout)
+	defer cancel()
+
+	om := object.NewOptionManager(c.Client.Client, *c.Client.Client.ServiceContent.Setting)
+	res, err := om.Query(ctx, "config.vpxd.stats.maxQueryMetrics")
+	if err == nil {
+		if len(res) > 0 {
+			if s, ok := res[0].GetOptionValue().Value.(string); ok {
+				v, err := strconv.Atoi(s)
+				if err == nil {
+					log.Printf("D! [input.vsphere] vCenter maxQueryMetrics is defined: %d", v)
+					if v == -1 {
+						// Whatever the server says, we never ask for more metrics than this.
+						return absoluteMaxMetrics, nil
```

Comment:
The only situation where this would kick in is if the setting in the config file is higher than ```absoluteMaxMetrics```. However, 100000 was way too high, so I lowered it to 10000.